Make weight conversion matcher more robust

hargup opened the pull request:
Summary of changes:

- Remove the requirement of "convert" in the query
- Match, `lb` and `kg`
- Refactor code for easier matching of synonyms.

sahil-shubham commented on apps/weight-converter/src/App.jsx:
```diff
@@ -106,9 +98,16 @@ const parseConversionString = (query) => {
-    let { amount, from, to } = parseConversionString(query);
-    from = mapUnits(from);
-    to = mapUnits(to);
+    let { amount, from, to } = parseConversionString(
+      replaceSynonyms(query, {
+        kilogram: ["kilograms", "kgs", "kg"],
+        milligram: ["milligrams", "mg", "mgs"],
+        tonne: ["tonnes", "tn", "ton"],
+        pound: ["pounds", "lbs", "lb"],
+        ounce: ["ounces", "oz"],
+        stone: ["stones", "st"],
+      })
+    );
```
Maybe we should include the keynames in the value array too, because this instant app doesn't work when it comes to query like "5 tonne to kilogram", i.e. when keynames are written.

sahil-shubham commented on apps/weight-converter/src/App.jsx:
```diff
@@ -106,9 +98,16 @@ const parseConversionString = (query) => {
-    to = mapUnits(to);
+    let { amount, from, to } = parseConversionString(
+      replaceSynonyms(query, {
+        kilogram: ["kilograms", "kgs", "kg"],
+        milligram: ["milligrams", "mg", "mgs"],
+        tonne: ["tonnes", "tn", "ton"],
+        pound: ["pounds", "lbs", "lb"],
+        ounce: ["ounces", "oz"],
+        stone: ["stones", "st"],
```
Also, "gram" is missing.

hargup replied:
Done!

sahil-shubham reviewed (approved):
Code looks better, thanks @hargup 

I suggested a few changes. Please give them a look.

hargup commented:
@N-Shar-ma

hargup commented:
@sahil-shubham addressed all the comments, also changed to `replaceSynonyms` to match full words instead of substring. For example, the earlier implementation would have matched `gram` in `anagram`, but not in present implementation.
